Core/WumpusCore/GameLocations/GameLocations.cs

```diff
@@ -51,26 +63,45 @@ public int GetEmptyRoom()
         }
 
         /// <summary>
-        /// Adds a room to the List.
+        /// Adds a room to the <c>rooms</c> list.
         /// </summary>
-        /// <param name="type">The type of room to add.</param>
+        /// <param name="type">The <c>RoomTypes</c> type of room to add.</param>
         /// <param name="pos">The position of the room to add.</param>
-        public void AddRoom(RoomTypes type, uint pos)
+        /// <exception cref="ArgumentException"></exception>
+        public void AddRoom(RoomTypes type, ushort pos)
         {
+            for (int i = 0; i < rooms.Count; i++)
+            {
+                if (pos == rooms[i].pos && rooms[i].type != RoomTypes.Flats)
+                {
+                    throw new ArgumentException();
```

Comment:
You could throw a better exception than this. Put a message in it! Also this is a very weird way to allow overwriting flats with other types. Maybe use a different method to change the type?

Reply:
Looks like you already have that method. In that case, why is this the way it is? Do you have another method that is blindly operating on rooms?

Reply:
I added the SetRoom method later. I'm not quite sure what I was thinking here, maybe I wanted to do the setting in the AddRoom?